include/zephyr/bluetooth/gatt.h

```diff
@@ -72,6 +72,18 @@ enum {
 	 *  passed to write callback.
 	 */
 	BT_GATT_PERM_PREPARE_WRITE = BIT(6),
+
+	/** @brief Attribute read permission with 128bit entropy key connection.
+	 *
+	 *  If set, read access requires connection using a 128bit entropy derived key.
```

Comment:
Could you add a "In practice this means.."? I ask for this because I'm unsure of what this PR is about. What does it mean that a connection key is derived from "128bit entropy"? The implementation of this check seems to pass if and only if LESC (`BT_KEYS_SC`) is used. Is that right?

Is the intention of `BT_GATT_PERM_READ_128B_ENTROPY` check that the connection has `BT_SECURITY_L4` (Level 4: Authenticated Secure Connections and 128-bit key.)?

Is the intention to check that the connection has a 128-bit key irrespective even if not authenticated?

Reply:
It seems to me that the spec explicitly requires that either LESC or OOB is used. That should probably be part of the flag's description. Is it possible to have a key with 128bit of entropy in some other way in Zephyr? If that is the case the name  128b_entropy would be inappropriate.

Reply:
>  Is it possible to have a key with 128bit of entropy in some other way in Zephyr

It is (currently) not possible to achieve 128bit of entropy without BR/EDR or LE Secure Connections or an OOB method in Bluetooth. 

So the security requirements that we want does fit with the name "128B_ENTROPY", but that may not be very obvious to many. 

I think it is imperative that we choose a name here that no only covers LESC, but possibly also OOB (unless we want that as separate flag), that is understandable. 

Furthermore, if we only check LESC with this flag, it is misleading (as OOB is equally valid), and as such we should at the very least add another issue for the OOB part, or add that as part of this PR.

Reply:
My understanding is that the requirement is about the key to be 128B entropy, and my interpretation is that the sources for getting the key (BR/EDR CS, LESC, OOB) are listed because these are the only ways to get such a key (OOB should cover everything else but the SCs, right?). To avoid legacy pairing thus becomes a side effect.
The question you pose @alwa-nordic whether there is any other way to achieve a 128B key in Zephyr is thus quite essential.

Regardless, its a good point about the description here, I will update it!